dask_ml/wrappers.py

```diff
@@ -0,0 +1,243 @@
+            return _apply_partitionwise(X, predict)
+
+        else:
+            return predict(X)
+
+    def predict_proba(self, X):
+        """Predict for X.
+
+        For dask inputs, a dask array or dataframe is returned. For other
+        inputs (NumPy array, pandas dataframe, scipy sparse matrix), the
+        regular return value is returned.
+
+        If the underlying estimator does not have a ``predict_proba``
+        method, then an ``AttributeError`` is raised.
+
+        Parameters
+        ----------
+        X : array or dataframe
+
+        Returns
+        -------
+        y : array-like
+        """
+
+        predict_proba = self._check_method('predict_proba')
+
+        if isinstance(X, da.Array):
+            # XXX: multiclass
+            return X.map_blocks(predict_proba,
+                                dtype='float',
+                                chunks=(X.chunks[0], len(self.classes_)))
+        elif isinstance(X, dd._Frame):
+            return _apply_partitionwise(X, predict_proba)
+        else:
+            return predict_proba(X)
+
+    def _check_method(self, method):
+        """Check if self.estimator has 'method'.
+
+        Raises
+        ------
+        AttributeError
+        """
+        if not hasattr(self.estimator, method):
+            msg = ("The wrapped estimator '{}' does not have a "
+                   "'{}' method.".format(self.estimator, method))
+            raise AttributeError(msg)
+        return getattr(self.estimator, method)
+
+
+def _first_block(dask_object):
+    """Extract the first block / partition from a dask object
+    """
+    if isinstance(dask_object, da.Array):
+        if dask_object.ndim > 1 and dask_object.numblocks[-1] != 1:
+            raise NotImplementedError("IID estimators require that the array "
+                                      "blocked only along the first axis. "
+                                      "Rechunk your array before fitting.")
+        shape = (dask_object.chunks[0][0],)
+        if dask_object.ndim > 1:
+            shape = shape + (dask_object.chunks[1][0],)
+
+        return da.from_delayed(dask_object.to_delayed().flatten()[0],
+                               shape,
+                               dask_object.dtype)
+
+    if isinstance(dask_object, dd._Frame):
+        return dask_object.get_partition(0)
+
+    else:
+        return dask_object
+
+
+def _apply_partitionwise(X, func):
+    """Apply a prediction partition-wise to a dask.dataframe"""
+    sample = func(X._meta_nonempty)
+    if sample.ndim <= 1:
+        p = ()
+    else:
+        p = (sample.shape[1],)
+
+    if isinstance(sample, np.ndarray):
+        blocks = X.to_delayed()
+        arrays = [
+            da.from_delayed(dask.delayed(func)(block),
+                            shape=(np.nan,) + p,
+                            dtype=sample.dtype)
+            for block in blocks
+        ]
+        return da.concatenate(arrays)
```

Comment:
Is this something that we would want to push upstream into Dask?

Reply:
Yeah, I could see it being broadly useful: https://github.com/dask/dask/issues/3138